File: STRINGS/344-reverse-string/reverse-string.js
var reverseString = function (s) {
    left = 0
    right = s.length - 1
    while (left < right) {
        if (s[left] !== s[right]) {
            [s[left], s[right]] = [s[right], s[left]];
        }
        left += 1;
        right -= 1;
    }
};

File: STRINGS/344-reverse-string/reverse-string.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int left = 0;
        int right = size(s) - 1;
        while (left < right) {
            if (s[left] != s[right]) {
                swap(s[left], s[right]);
            }
            left += 1;
            right -= 1;
        }
    }
};